// @geist-ui/react/esm/col/col.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import _JSXStyle from "styled-jsx/style";
import React from 'react';
import withDefaults from '../utils/with-defaults';
var defaultProps = {
  span: 24,
  offset: 0,
  component: 'div',
  className: ''
};

var Col = function Col(_ref) {
  var component = _ref.component,
      children = _ref.children,
      span = _ref.span,
      offset = _ref.offset,
      className = _ref.className,
      props = _objectWithoutProperties(_ref, ["component", "children", "span", "offset", "className"]);

  var Component = component;
  return /*#__PURE__*/React.createElement(Component, _extends({}, props, {
    className: _JSXStyle.dynamic([["540733280", [100 / 24 * span, 100 / 24 * offset]]]) + " " + (props && props.className != null && props.className || "col ".concat(className))
  }), children, /*#__PURE__*/React.createElement(_JSXStyle, {
    id: "540733280",
    dynamic: [100 / 24 * span, 100 / 24 * offset]
  }, ".col.__jsx-style-dynamic-selector{float:left;box-sizing:border-box;padding-left:calc(var(--row-gap) / 2);padding-right:calc(var(--row-gap) / 2);width:".concat(100 / 24 * span, "%;margin-left:").concat(100 / 24 * offset, "%;}")));
};

var MemoCol = /*#__PURE__*/React.memo(Col);
export default withDefaults(MemoCol, defaultProps);
