// @geist-ui/react/esm/auto-complete/auto-complete-dropdown.js
import _JSXStyle from "styled-jsx/style";
import React from 'react';
import useTheme from '../use-theme';
import withDefaults from '../utils/with-defaults';
import { useAutoCompleteContext } from './auto-complete-context';
import Dropdown from '../shared/dropdown';
var defaultProps = {
  className: '',
  dropdownStyle: {}
};

var AutoCompleteDropdown = function AutoCompleteDropdown(_ref) {
  var children = _ref.children,
      visible = _ref.visible,
      className = _ref.className,
      dropdownStyle = _ref.dropdownStyle,
      disableMatchWidth = _ref.disableMatchWidth;
  var theme = useTheme();

  var _useAutoCompleteConte = useAutoCompleteContext(),
      ref = _useAutoCompleteConte.ref;

  var clickHandler = function clickHandler(event) {
    event.preventDefault();
    event.stopPropagation();
    event.nativeEvent.stopImmediatePropagation();
  };

  return /*#__PURE__*/React.createElement(Dropdown, {
    parent: ref,
    visible: visible,
    disableMatchWidth: disableMatchWidth
  }, /*#__PURE__*/React.createElement("div", {
    style: dropdownStyle,
    onClick: clickHandler,
    className: _JSXStyle.dynamic([["705864980", [theme.layout.radius, theme.expressiveness.shadowLarge, theme.palette.background]]]) + " " + "auto-complete-dropdown ".concat(className)
  }, children, /*#__PURE__*/React.createElement(_JSXStyle, {
    id: "705864980",
    dynamic: [theme.layout.radius, theme.expressiveness.shadowLarge, theme.palette.background]
  }, ".auto-complete-dropdown.__jsx-style-dynamic-selector{border-radius:".concat(theme.layout.radius, ";box-shadow:").concat(theme.expressiveness.shadowLarge, ";background-color:").concat(theme.palette.background, ";overflow-y:auto;max-height:15rem;overflow-anchor:none;}"))));
};

export default withDefaults(AutoCompleteDropdown, defaultProps);
